### PR TITLE
Removed the broken caching of getMean() and provided a slightly more …

### PR DESCRIPTION
**Pull request by tfenne**
…efficient implementation of getMean() that didn't require traversing all the bins twice.

### CODE REVIEW

**Comment by tfenne** [comment]
Solution for issue #347

File: src/java/htsjdk/samtools/util/Histogram.java [hunk 1]
```diff
@@ -164,12 +161,21 @@ public boolean equals(final Object o) {
...
     }
 
+    /**
+     * Assuming that the key type for the histogram is a Number type, returns the mean of
+     * all the items added to the histogram.
+     */
     public double getMean() {
-        if (mean == null) {
-            mean = getSum() / getCount();
+        double product=0, totalCount=0;
```
Comment (yfarjoun) [inline src/java/htsjdk/samtools/util/Histogram.java]: spaces

**Comment by yfarjoun** [comment]
looks  fine to me. Could you add a comment explaining why you aren't using getSum and getCounts? also, the test uses round(), could you change that to use the (newish) double comparator that takes a precision?

**Comment by tfenne** [comment]
@yfarjoun: added the requested comment.  I couldn't find the comparator you mentioned - the only thing I could find was `TestNGUtil.compareDoubleWithAccuracy()` but that's up in picard and not available here.

**Comment by yfarjoun** [comment]
:100:

**Pull request merged**